be/src/vec/exec/scan/vfile_scanner.cpp

```diff
@@ -256,7 +256,20 @@ Status VFileScanner::_fill_columns_from_path(size_t rows) {
                 ss << "Unknown source slot descriptor, slot_id=" << slot_desc->id();
                 return Status::InternalError(ss.str());
             }
-            const std::string& column_from_path = range.columns_from_path[it->second];
+
+            // For external table query, find the index of column in path.
+            // Reason is _partition_slot_index_map may be incorrect for query
+            // Because query doesn't always search for all columns in a table and the select columns order is random.
+            vector<std::string>::const_iterator iit =
```

Comment:
Better do this in prepare phase once. Otherwise, we have to find keys for each `_fill_columns_from_path()` call, which is inefficient.
You can use the structure to same the mapping from `_partition_slot_descs` to the index of `columns_from_path_keys`